data/hai/hai reveal 1 intro.txt

```diff
@@ -0,0 +1,1002 @@
+# Copyright (c) 2020 MasterOfGrey
+#
+# Endless Sky is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+########################################################################
+
+mission "Hai Secret Leaker Begins"
+	landing
+	invisible
+	to offer
+		has "main plot completed"
+	on offer
+		event "pug media coverage" 20 30
+		fail
+
+event "pug media coverage"
+	# add attitudes for aliases of governments
+	government "Republic"
+		"attitude toward"
+			"Republic that won't enter wormhole" 1.0
+	government "Republic that won't enter wormhole"
+		"attitude toward"
+			"Republic" 1.0
+
+	government "Syndicate"
+		"attitude toward"
+			"Syndicate that won't enter wormhole" 1.0
+	government "Syndicate that won't enter wormhole"
+		"attitude toward"
+			"Syndicate" 1.0
+
+	government "Free Worlds"
+		"attitude toward"
+			"Free Worlds that won't enter wormhole" 1.0
+	government "Free Worlds that won't enter wormhole"
+		"attitude toward"
+			"Free Worlds" 1.0
+
+	government "Deep Security"
+		"attitude toward"
+			"Deep Security that won't enter wormhole" 1.0
+	government "Deep Security that won't enter wormhole"
+		"attitude toward"
+			"Deep Security" 1.0
+
+event "hai ladybug available"
+	shipyard "Hai Intermediate"
+		"Ladybug"
+	shipyard "Hai Advanced"
+		"Ladybug"
+
+mission "Hai Reveal: Reputation Reset"
+	landing
+	invisible
+	to offer
+		has "event: pug media coverage"
+	on offer
+		# Sync reputations for aliases of governments:
+		"reputation: Deep Security that won't enter wormhole" = "reputation: Deep Security"
+		"reputation: Republic that won't enter wormhole" = "reputation: Republic"
+		"reputation: Free Worlds that won't enter wormhole" = "reputation: Free Worlds"
+		"reputation: Syndicate that won't enter wormhole" = "reputation: Syndicate"
+
+		# Make sure the new governments start with appropriate rep:
+		"reputation: Human Alliance that won't enter wormhole" = 1000
+		"reputation: Human Alliance" = 1000
+		"reputation: Hai (Friendly Unfettered)" = 1000
+		"reputation: Bounty Hunter that Attacks Hai" = -1000
+		"reputation: Philosophers" = 1000
+		"reputation: Pirate (Devil-Run Gang)" = -1000
+		fail
+
+mission "Event: Pug Media Coverage"
+	landing
+	source
+		government "Republic" "Syndicate" "Free Worlds"
+		attributes "spaceport"
+	to offer
+		has "event: pug media coverage"
+	on offer
+		event "hai secret leaks" 10 15
```

Comment:
Given that some of the dialog seems contingent on the main campaign having occurred recently, but an old pilot updating to a version that includes this content could have spent years past the main campaign, might we want to move the trigger for all this to the final mission of the main campaign instead of an invisible mission that triggers if the main campaign is done? If this is merged alongside a rework of the FW campaign, I imagine it could easily be justifiable to lock all this to new pilots.

Reply:
IMO the events really only make sense as an immediate follow on. If we extended the trigger it would be purely for gameplay reasons, not anything else.
I'm also inclined to say that old pilots *expect* new content might have potentially weird time-gaps. That's just a feature of games updating. I thought the whole point was for ES to be backwards compatible, so people didn't have to start new games to experience new content that wouldn't normally have been locked out? This suggestion seems to counter that philosophy.

Reply:
We've had lockouts on new content in the past, but those were typically as a result of minor additions to existing campaigns as opposed to entire new storylines. The biggest lockout that comes to mind is the second half of the Terraforming missions, where players who had completed Terraforming Rand before Terraforming Tundra couldn't do Terraforming Tundra without a new pilot (or save editing, of course), and the reason for that was explicitly because of the lead-in to Terraforming Tundra being time-sensitive relative to when you completed Terraforming Rand.

And I'd say the events only make sense as an immediate follow up insofar as that's how you've written it. I don't see why the intro couldn't be rewritten to occur some time later. And the gameplay reasons for why we would extend the trigger time is entirely what my concern is. As of right now once you finish the main campaign the world is opened up to you, and there are already a number of existing missions that can only be done after the main campaign is complete. By having these events occur so quickly after the main campaign while also having no way to story wise break out of them (no defers or declines as far as I've seen), you suffocate out those other missions, namely the FW epilogues (not too important, although I imagine those epilogues would change when you meet people who you met during this storyline) and Emerald Sword missions. In the latter case it seems to especially be an issue to me that the events of Hai Reveal lead you right to the Wanderers (as far as I've seen from reading your comments elsewhere), which I'd say even further devalues the Emerald Sword by opening up the player to T2 tech before giving them a real chance to get their hands on the Emerald Sword. And should there be any future post-campaign content that's still contained within human space, this may crowd those out too.

I also take issue with this leading to the Wanderers in that it takes away from the exploration aspect of the game by pointing you to where you "need" to go. (I put need in quotes because technically a player could stall completing any mission at any time and go explore elsewhere, but simply having a mission pressures the player into doing that mission, thereby creating a perceived need to do things a certain way.) As mentioned, after beating the main campaign the game is opened up to you and you can go where ever you want with your new found jump drive, but these missions, in being forced (i.e. can't be deferred or declined) and so immediate after the main campaign, create a continuous pipeline from starting the main campaign up to beating it, starting Hai reveal and beating it, then starting the Wanderer campaign, leaving little room to explore elsewhere, such as exploring the Coalition or Remnant, and basically handing you the Wanderer campaign instead of having it be something you stumble upon as is done now.

Reply:
I understand your concerns, though I do have a few points:

> I don't see why the intro couldn't be rewritten to occur some time later.

I don't see *how* it *could* be rewritten to occur some time later. It takes advantage of very specific conditions that would only realistically be in place for a short time.

> you suffocate out those other missions, namely the FW epilogues

To be honest, those could (and some of them possibly should) just be moved to after this.

> and Emerald Sword missions

I don't recall those actually being all that easy to trip over tbh, and the Emerald Sword is a spectacular unique flagship one way or another. To me this seems like a non-issue.

> In the latter case it seems to especially be an issue to me that the events of Hai Reveal lead you right to the Wanderers
etc.

Eh, sort of? But also not really. It hints that the Unfettered have something going on with another race, and then once it's all over and done with, if you don't go to the wanderers for a decent length of time, there's a "You know you saw something that pointed to this section of space" type nudge. There's also that a lot of the community consider the Wanderers to be "the main plot part 2" and at present it's *very* disconnected, and lets you start it kind of illogically before the main campaign, but then hangs up in the middle without explanation until you do the main campaign. As far as the Wanderer's plot is concerned all this does is provide some more structure to the narratives leading up to and surrounding it.

> which I'd say even further devalues the Emerald Sword by opening up the player to T2 tech before giving them a real chance to get their hands on the Emerald Sword

And again, the Emerald sword isn't valued for what Tier it is, it's valued entirely independent of that.